corehq/apps/userreports/models.py

```diff
@@ -265,15 +265,25 @@ def all(cls):
 
 
 class CustomDataSourceConfiguration(JsonObject):
+    _datasource_id_prefix = 'custom-'
     domains = ListProperty()
     config = DictProperty()
 
     @classmethod
-    def all(cls):
+    def all(cls, domain=None):
```

Comment:
is a bit subjective but I think i would have gone with a `by_domain` classmethod that just calls `.all()` and filters the results. that's more conventional in other parts of HQ, as well as consistent with the `DataSourceConfiguration` API